harness/determined/util.py

```diff
@@ -246,6 +248,37 @@ def jsonable(obj: Any) -> Any:
     return json.dumps(jsonable(obj), indent=indent, sort_keys=sort_keys)
 
 
+def _make_shutil_ignore(root_path: pathlib.Path) -> Callable:
+    ignore = list(common_constants.DEFAULT_DETIGNORE)
+    ignore_path = root_path / ".detignore"
+    if ignore_path.is_file():
+        with ignore_path.open("r") as detignore_file:
+            ignore.extend(detignore_file)
+
+    import pathspec
```

Comment:
why the non-standard import?

Reply:
nvm, I see it in Anda's old lazy imports to speedup library import time.  Must have been slow in some benchmarking.

Reply:
that might be worth a comment